src/Test.Utilities/DiagnosticAnalyzerTestBase.cs

```diff
@@ -390,19 +418,23 @@ protected static Project CreateProject(string[] sources, string language = Langu
                 .AddMetadataReference(projectId, SystemRuntimeFacadeRef)
                 .AddMetadataReference(projectId, SystemThreadingFacadeRef)
                 .AddMetadataReference(projectId, SystemThreadingTaskFacadeRef)
-                .AddMetadataReference(projectId, s_immutableCollectionsReference)
                 .AddMetadataReference(projectId, s_workspacesReference)
                 .AddMetadataReference(projectId, s_systemDiagnosticsDebugReference)
                 .AddMetadataReference(projectId, s_systemDataReference)
                 .WithProjectCompilationOptions(projectId, options)
                 .GetProject(projectId);
 
-            if (addLanguageSpecificCodeAnalysisReference)
+            if ((referenceFlags & ReferenceFlags.RemoveCodeAnalysis) != ReferenceFlags.RemoveCodeAnalysis)
```

Comment:
If the default is `ReferenceFlags.None`, will that mean majority of the tests will end up with an unnecessary reference to CodeAnalysis?

Reply:
They already do, the default for addLanguageSpecificCodeAnalysisReference is true today. https://github.com/dotnet/roslyn-analyzers/blob/master/src/Test.Utilities/DiagnosticAnalyzerTestBase.cs#L422. I actually broke a bunch of tests during initial implementation of this because I assumed that the default was false without actually checking.

Reply:
That seems unfortunate, but unrelated to your PR.